File: leetcode/Question-1.js
// Question 1 Find index of sum of two element

let arr = [2, 7, 8, 9];
let target = 9;

function findindexofsum(array, target) {
    let result = {};

    for (let i = 0; i < array.length; i++) {
        let currentNumber = array[i];
        let need = target - currentNumber;
        // need = 9 - 2 = 7;
        if (result[need] !== undefined) {
            return [result[need], i];
        }

        result[currentNumber] = i;
    }
}

let response = findindexofsum(arr, target);
if (response === undefined) {
    console.log("Element Is Note Found!! To Sum Of Target Elements !");
}
else {
    console.log(response);
}

File: leetcode/Question-1.cpp
// Question 1 Find index of sum of two element
#include <iostream>
using namespace std;

int main()
{
    int arr[] = {2, 7, 8, 9};
    int target = 9;
    int n = 4;

    for (int i = 0; i < n; i++)
    {
        for (int j = i + 1; j < n; j++)
        {
            if (arr[i] + arr[j] == target)
            {
                cout << i << " " << j;
                return 0;
            }
        }
    }

    cout << "Not Found";
    return 0;
}
